tests/aggregation_regress/tests.py

```diff
@@ -554,6 +554,10 @@ def test_conditional_aggregate_on_complex_condition(self):
             325,
         )
 
+    def test_q_annotation_aggregate(self):
+        # Ensure aggregation works with annotations using Q objects
+        self.assertEqual(Book.objects.annotate(has_pk=~Q(pk=None)).count(), 6)
```

Comment:
Negation is unnecessary here so I would simplify this a bit:
```suggestion
        self.assertEqual(Book.objects.annotate(has_pk=Q(pk__isnull=False)).count(), 6)
```

Reply:
I agree with the change but disagree with the reasoning 🙂

I don't think `~Q(…)` is any more or less "necessary" or simple than its equivalent `Q(…__isnull)` … they're both about the same in terms of readability/necessity/simplicity/whatever.

In tests I usually recommend that folks go for the more readable/maintainable and that the people who's responsibility it is to maintain it get more of a say. In this case you prefer the `isnull` variant so that's what should be chosen as you & Carlton are the current Django fellows… and its your responsibility to maintain this. ☺️